synapseclient/client.py

```diff
@@ -1354,6 +1355,130 @@ def uploadFileHandle(self, path, parent, synapseStore=True, mimetype=None, md5=N
+        :param separator:       The delimiter to be used for separating entries in the resulting file. Defaults to ",".
+        :param header:          Is the first line a header? Defaults to True.
+
+        :returns: Filehandle of download list manifest
+        """
+        request_body = {
+            "concreteType": "org.sagebionetworks.repo.model.download.DownloadListManifestRequest",
+            "csvTableDescriptor": {
+                "separator": separator,
+                "quoteCharacter": quoteCharacter,
+                "escapeCharacter": escapeCharacter,
+                "lineEnd": lineEnd,
+                "isFirstLineHeader": header
+            }
+        }
+        return self._waitForAsync(uri="/download/list/manifest/async", request=request_body)
+
+    def get_download_list_manifest(self):
+        """Get the path of the download list manifest file
+
+        :returns: path of download list manifest file
+        """
+        manifest = self._generate_manifest_from_download_list()
+        # Get file handle download link
+        file_result = self._getFileHandleDownload(
+            fileHandleId=manifest['resultFileHandleId'],
+            objectId=manifest['resultFileHandleId'],
+            objectType="FileEntity"
+        )
+        # Download the manifest
+        downloaded_path = self._download_from_URL(
+            url=file_result['preSignedURL'],
+            destination="./",
+            fileHandleId=file_result['fileHandleId'],
+            expected_md5=file_result['fileHandle'].get('contentMd5')
+        )
+        return downloaded_path
+
+    def get_download_list(self, manifest=True, downloadLocation=None) -> str:
+        """Download all files from your Synapse download list
+
+        :param manifest: Whether to keep the manifest file for the download list. Defaults to True.
+        :param downloadLocation: Directory to download files to.
+
+        :returns: manifest file with file paths
+        """
+        dl_list_path = self.get_download_list_manifest()
+        downloaded_files = []
+        with open(dl_list_path) as manifest_f, \
+             open('SYNAPSE_METADATA_MANIFEST.tsv', 'w') as write_obj:
+
+            reader = csv.DictReader(manifest_f)
+            columns = reader.fieldnames
+            columns.append("path")
+            # Write the downloaded paths to a new manifest file
+            writer = csv.DictWriter(write_obj, fieldnames=columns, delimiter='\t')
+            writer.writeheader()
+
+            for row in reader:
+                # You can add things to the download list that you don't have access to
+                # So there must be a try catch here
+                try:
+                    entity = self.get(row['ID'], downloadLocation=downloadLocation)
+                    downloaded_files.append(entity.path)
+                    row['path'] = entity.path
+                    writer.writerow(row)
```

Comment:
Create a list of successfully downloaded files then all remove from download list at the very end after all downloads will succeed

Reply:
Unsuccessful downloads should not be in the manifest file